examples/action_items/index.ts

```diff
@@ -0,0 +1,87 @@
+import Instructor from "@/instructor"
+import OpenAI from "openai"
+import { z } from "zod"
+
+const PrioritySchema = z.enum(["HIGH", "MEDIUM", "LOW"]);
+
+const SubtaskSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+})
+
+const TicketSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+  description: z.string(),
+  priority: PrioritySchema,
+  assignees: z.array(z.string()),
+  subtasks: z.array(SubtaskSchema).optional(),
+  dependencies: z.array(z.number()).optional()
+})
+
+const ActionItemsSchema = z.object({
+  items: z.array(TicketSchema)
+})
+
+type ActionItems = z.infer<typeof ActionItemsSchema>
+
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+  organization: process.env.OPENAI_ORG_ID ?? undefined
+})
+
+const client = Instructor({
+  client: oai,
+  mode: "JSON",
+})
+
+const extractActionItems = async (data: string): Promise<ActionItems | undefined> => {
```

Comment:
can we make this `async def` is that better? do we need the undefined?

Reply:
What do you mean with `async def`? `client.chat.completions.create` has return type `any`, I am not sure if it ever returns a falsy value / what happens in case of an error but i think it might make sense to return undefined if the extraction wasn't successful (though then it might just raise an error)